### PR TITLE
feat: can now pass parameters to jobs

### PR DESCRIPTION
**Pull request by YaniKolev**
Signed-off-by: Yani Kolev <yani.kolev@tngtech.com>

### CODE REVIEW

File: test/executor/JobExecutor.spec.ts [hunk 1]
```diff
@@ -88,20 +90,4 @@ describe('JobExecutor', () => {
     verify(executionsRepository.addExecution(scheduleId, job.name, job.maxRunning)).once();
     verify(executionsRepository.removeExecution(scheduleId, job.name)).once();
   });
-
-  it('writes result to mongo', async () => {
```
Comment (YaniKolev) [inline test/executor/JobExecutor.spec.ts:92]: This tests was just  a clone of one of the previous one.

File: src/schedule/Schedule.ts [hunk 2]
```diff
@@ -87,10 +92,11 @@ export class Schedule extends LogEmitter {
    * You might not want to await the promise if your specified delay is big.
    *
    * @param name the job to run
+   * @param parameters the parameters that will be passed to the job for this single run; passing nothing will run the job with no parameters, NOT the parameters used for the scheduled runs
```
Comment (kress13013) [inline src/schedule/Schedule.ts:95]: Just for clarification: with "NOT the parameters used for the scheduled runs" you want to say that they are not used as fallback in case no parameter would be provided?
Maybe sth like `passing nothing will run the job with no parameters, regardless of potentially set job parameters for the scheduled runs`?

Comment (YaniKolev) [reply src/schedule/Schedule.ts:95]: Exactly. Yeah, that sounds better!

**Review by kress13013 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -63,6 +73,8 @@ const mongoSchedule = new MongoScheduleBuilder()
   .withJob(cronJob)
   .build();
 
+mongoSchedule.define(cronJobWithNoParameters);
```
Comment (NiklasEi) [inline README.md:76]: Why do we define this job, but not the others?

File: README.md [hunk 4]
```diff
@@ -106,25 +119,25 @@ Only the job with the provided name is started/stopped/cancelled/removed.
...
+| clear     |                                                                             | Removes all jobs from the database. This also removes jobs that are not on this schedule, but were defined by other schedules. However, does NOT stop job executions - this will cause currently running jobs to fail. Consider using stop/cancel/remove methods instead!                             |
+| get       | `string`                                                                    | Returns a description of the job. Returns undefined if no job with the provided name is defined.                                                                                                                                                                                                      |
+| run       | `string`,  `JobParameters` (default: `undefined`),  `number` (default: `0`) | Runs the job with the provided name once, independently from the schedule, after the specified delay.  If no parameters are provided, the job will be called with no parameters.  Note that `maxRunning` is respected, ie. the execution is skipped if the job is already running `maxRunning` times. |
```
Comment (NiklasEi) [inline README.md:139]: The number param isn't documented. Could you add another sentence about it?

**Review by NiklasEi — CHANGES_REQUESTED** [review]
I am missing some tests where the job handler is asserting the job parameters (or `undefined` if none where given).

**Review by NiklasEi — APPROVED** [review]
